### ISSUE TITLE
LibSOE Revamp

### ISSUE THREAD

**Issue by SengokuNadeko** [OpenFreeRealms/OpenFreeRealms#3]
# TODO
- Rewrite LibSOE to handle packets properly
- Implement packet structures to the LibSOE

# Completed
- Solved the issue of out of order acknowledgement packets being sent. Could use some polishing.

**Comment by Gmjjr** [comment]
should probably provide details.
issues aren't much help if they don't explain what's wrong

**Comment by SengokuNadeko** [comment]
yup just did it, i may add more to it later depending on what other problems we find.
